R/plot.R

```diff
@@ -448,46 +453,92 @@ contour.stars = function(x, ...) {
 #' if (require(ggplot2)) {
 #'  ggplot() + geom_stars(data = r) + scale_fill_identity()
 #' }
-st_rgb = function(x, dimension = 3, use_alpha = dim(x)[dimension] == 4, maxColorValue = 255L, 
-		probs = c(0., 1.), stretch = FALSE) {
+#' r = st_rgb(x[,,,3:1],
+#' 		   probs = c(0.01, 0.99),
+#' 		   stretch = "percent")
+#' plot(r)
+#' r = st_rgb(x[,,,3:1],
+#' 		   probs = c(0.01, 0.99),
+#' 		   stretch = "histogram")
+#' plot(r)
+st_rgb <- function (x,
+					dimension = 3,
+					use_alpha = dim(x)[dimension] == 4,
+					maxColorValue = 255L,
+					probs = c(0, 1),
+					stretch = NULL) {
 	if (is.character(dimension))
 		dimension = match(dimension, names(dim(x)))
-	stopifnot(is.numeric(dimension), length(dimension)==1)
-	if (!dim(x)[dimension] %in% c(3,4))
-		stop(paste("number of bands along dimension", dimension, "should be 3 or 4"))
+	stopifnot(is.numeric(dimension), length(dimension) == 1)
+	if (!dim(x)[dimension] %in% c(3, 4))
+		stop(paste("number of bands along dimension", dimension,
+				   "should be 3 or 4"))
 	dims = setdiff(seq_along(dim(x)), dimension)
-	cutoff = function(x, probs) {
-		qs = if (all(probs == c(0., 1.)))
+	cutoff = function(x, probs, stretch.method = "percent") {
+		if(stretch.method == "percent"){
+			qs = if (all(probs == c(0, 1)))
 				range(x)
-			else
-				quantile(x, probs, na.rm = TRUE)
-		x = (x - qs[1])/(qs[2] - qs[1])
-		x[x > 1] = 1
-		x[x < 0] = 0
-		x * maxColorValue
+			else quantile(x, probs, na.rm = TRUE)
+			x = (x - qs[1])/(qs[2] - qs[1])
+			x[x > 1] = 1
+			x[x < 0] = 0
+			x * maxColorValue
+		} else if(stretch.method == "histogram"){
+			x = stats::ecdf(x)(x)
+			x * maxColorValue
+		} else {
+			x
+		}
 	}
-	if (stretch)
-		x = st_apply(x, dimension, cutoff, probs = probs)
+
+	if(is.null(stretch)) {
+		stretch.method = "none"
+		stretch = FALSE
+	}
+
+	if(is.logical(stretch)) {
+		if(stretch){
+			stretch.method = "percent"
+			message("Stretch as logical == TRUE, then percent clip by probs is applied")
```

Comment:
I don't think a message is needed if the users sets a parameter; it should be obvious from the documentation what the user wants when [s]he sets this parameter.

Reply:
Ok, that make sense. I will remove these lines